WordPress/Classes/Models/Blog.m

```diff
@@ -292,9 +293,34 @@ - (NSString *)postFormatTextFromSlug:(NSString *)postFormatSlug
 // WP.COM private blog.
 - (BOOL)isPrivate
 {
-    return (self.isHostedAtWPcom && [[self getOptionValue:@"blog_public"] isEqualToString:@"-1"]);
+    return (self.isHostedAtWPcom && [self.privacy isEqualToNumber:@(BlogPrivacyPrivate)]);
 }
 
+- (NSString *)textForBlogPrivacy:(BlogPrivacy)privacy
+{
+    if (!self.privacy) {
+        return NSLocalizedString(@"Unknow", @"Text for unknow privacy setting");
```

Comment:
This check should (potentially) live within `textForCurrentBlogPrivacy`, otherwise the method might not work as expected, since it's return value is tied up to `self.privacy`'s value